cmds/core/ip/ops.go

```diff
@@ -130,3 +131,66 @@ func showNeighbours(w io.Writer, withAddresses bool) error {
 	}
 	return nil
 }
+
+const (
+	defaultFmt = "default via %v dev %s proto %s metric %d\n"
+	routeFmt   = "%v dev %s proto %s scope %s src %s metric %d\n"
+)
+
+var rtProto = map[int]string{
+	unix.RTPROT_BOOT:   "boot",
+	unix.RTPROT_KERNEL: "kernel",
+	unix.RTPROT_STATIC: "static",
+	unix.RTPROT_DHCP:   "dhcp",
+}
+
+func showRoutes(inet6 bool) error {
+	var f int
+	if inet6 {
+		//TODO: implement ipv6 implementation
+		//f = netlink.FAMILY_V6
+		return fmt.Errorf("ipv6 route not implemented yet")
+	}
+	f = netlink.FAMILY_V4
+
+	routes, err := netlink.RouteList(nil, f)
+	if err != nil {
+		return err
+	}
+	for _, route := range routes {
+		if route.Gw != nil {
+			err = defaultRoute(route)
+		} else {
+			err = showRoute(route)
+		}
+	}
+	return err
+}
+
+func defaultRoute(r netlink.Route) error {
+	l, err := netlink.LinkByIndex(r.LinkIndex)
```

Comment:
there's some duplication: you could move this call one level above, and pass the `netlink.Link` object here and to `showRoute`. At that point I'm not entirely sure if two separate functions are needed though

Reply:
your right. will be improved